### PR TITLE
Adds causalimpact recipe

### PR DESCRIPTION
**Pull request by jamalsenouci**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by nicornk** [comment]
hi @jamalsenouci, thanks for your work on causalimpact!
are you still working on this? Is there any way I can contribute to get causalimpact into conda-forge?

**Comment by jamalsenouci** [comment]
@nicornk I would love some help with this. When I did this initially I ran into a few build errors which just needed to be worked through. I'm happy to support if you want to lead it

**Comment by nicornk** [comment]
I have no push access to your fork https://github.com/jamalsenouci/staged-recipes/tree/71f3aba2d46e98921553af7c75b2fca068285e25/recipes/causalimpact

The logs of the previous attempt have been deleted already.

Can you grant me access to your fork so I can force push to retrigger a new build?

**Comment by jamalsenouci** [comment]
that should be granted now

**Comment by jamalsenouci** [comment]
@nicornk thanks for your work on this, I have had a quick attempt at updating causalimpact to use the latest version of pymc so that we can avoid the pinning on older versions. Now I just need to hope the build goes green again with these changes

**Comment by nicornk** [comment]
@conda-forge/staged-recipes Please review, thank you.

File: recipes/causalimpact/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools-scm
+  run:
+    - importlib-metadata  # [py<38]
```
Comment (ocefpaf) [inline recipes/causalimpact/meta.yaml:23]:
```suggestion
    - importlib-metadata
```

File: recipes/causalimpact/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  skip: true  # [py<36]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (ocefpaf) [inline recipes/causalimpact/meta.yaml:20]:
```suggestion
    - python >=3.6
```

File: recipes/causalimpact/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    - setuptools-scm
+  run:
+    - importlib-metadata  # [py<38]
+    - python
```
Comment (ocefpaf) [inline recipes/causalimpact/meta.yaml:24]:
```suggestion
    - python >=3.6
```

**Review by ocefpaf — COMMENTED** [review]
This can be noarch using the "trick" to install importlib-metadata py>38.

**Comment by nicornk** [comment]
@ocefpaf thx, done :)

**Pull request merged**